.github/workflows/build-applications.yml

```diff
@@ -48,20 +45,24 @@ jobs:
           - idf_ver: "release-v5.3"
             parallel_count: 2
             parallel_index: 2
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
     container: espressif/idf:${{ matrix.idf_ver }}
     steps:
-      - uses: actions/checkout@v3
-        with:
-          submodules: 'recursive'
-      - name: Build ESP-BSP examples
+      - uses: actions/checkout@v4
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v45
+
+      - name: Build ESP-BSP applications
         shell: bash
         env:
-          extra_arguments: "${{ matrix.idf_ver == 'latest' && '--default-build-targets esp32 esp32s2 esp32s3 esp32c2 esp32c3 esp32c6 esp32h2 esp32p4' || '' }}"
+          IDF_EXTRA_ACTIONS_PATH: "${{ github.workspace }}/examples"
+          ALL_CHANGED_FILES: ${{ steps.changed-files.outputs.all_changed_files }}
         run: |
-          export IDF_EXTRA_ACTIONS_PATH=${GITHUB_WORKSPACE}/examples
           . ${IDF_PATH}/export.sh
-          pip install idf-component-manager==1.* ruamel.yaml idf-build-apps --upgrade
-          echo idf-build-apps extra arguments: ${{ env.extra_arguments }}
-          idf-build-apps find ${{ env.extra_arguments }} -p examples
-          idf-build-apps build --parallel-count ${{ matrix.parallel_count }} --parallel-index ${{ matrix.parallel_index }} ${{ env.extra_arguments }} -p examples
+          pip install idf-component-manager==1.* ruamel.yaml idf-build-apps==2.4.3 --upgrade
+
+          echo "Files changed: ${ALL_CHANGED_FILES}"
+          idf-build-apps find --modified-files "${ALL_CHANGED_FILES}"
```

Comment:
no need to find first.

Reply:
Thanks! I like to have the output printed in the CI, just for me to double check that all jobs that I want are selected :)